app/src/routes/index.tsx

```diff
@@ -34,92 +44,102 @@ const UserRoute = () => {
 };
 
 const PermissionsRoute = (props: any) => {
-  const fbContext = useContext(FirebaseContext);
-
-  if (props?.validator && props.validator(fbContext?.role)) {
+  const { currentUser } = useAuth();
+  if (props?.validator && props.validator(currentUser?.role)) {
     return <Outlet />;
   }
   return <Error404 />;
 };
 
 const router = createBrowserRouter([
   {
-    path: "/",
-    element: <UserRoute />,
-    errorElement: <Error404 />,
+    element: <AuthLayout />,
+    errorElement: <Error404/>,
     children: [
       {
-        index: true,
-        element: <HomeRoute />,
-      },
-      {
-        path: "/home",
-        element: <HomeRoute />,
-      },
-      {
-        path: "/borrow",
-        element: <BorrowRoute />,
-        loader: async ({ request }) => {
-          const url = new URL(request.url);
-          const qid = url.searchParams.get("q");
-          if (!qid) {
-            return null;
-          }
+        path: "/",
+        element: <UserRoute />,
+        errorElement: <Error404 />,
+        children: [
+          {
+            index: true,
+            element: <HomeRoute />,
+          },
+          {
+            path: "/home",
+            element: <HomeRoute />,
+          },
+          {
+            path: "/borrow",
+            element: <BorrowRoute />,
+            loader: async ({ request }) => {
+              const url = new URL(request.url);
+              const qid = url.searchParams.get("q");
```

Comment:
highly recommend renaming `qid` and `tid`, I'm not sure what they're supposed to be

Reply:
qid is the id of the qr code, and tid is the transaction id I believe. I think we should rename them in a future pr, @royayush1